modules/actions/pool/sync-join-exits.ts

```diff
@@ -52,60 +52,42 @@ export const syncJoinExits = async (vaultSubgraphClient: V3VaultSubgraphClient,
-            // 🚨 Reading from the DB in a loop – will get slow with a large events volume
-            const prices = await prisma.prismaTokenPrice.findMany({
-                where: {
-                    tokenAddress: { in: event.pool.tokens.map((token) => token.address) },
-                    timestamp: roundToHour(Number(event.blockTimestamp)), // 🚨 Assuming all prices are available hourly
-                    chain: chain,
-                },
-                include: {
-                    token: true,
-                },
-            });
-
-            const usd = event.pool.tokens.map((token) => {
-                const price = prices.find((price) => price.tokenAddress === token.address);
-                return {
-                    address: token.address,
-                    amount: event.amounts[token.index],
-                    valueUSD:
-                        Number(formatUnits(BigInt(event.amounts[token.index]), price?.token?.decimals ?? 18)) *
-                        (price?.price || 0), // TODO: check USD amount
-                };
-            });
-
-            return {
-                id: event.id, // tx + logIndex
-                tx: event.transactionHash,
-                type: event.type === 'Join' ? PoolEventType.JOIN : PoolEventType.EXIT,
-                poolId: event.pool.id,
-                chain: chain,
-                vaultVersion,
-                userAddress: event.user.id,
-                blockNumber: Number(event.blockNumber),
-                blockTimestamp: Number(event.blockTimestamp),
-                logIndex: Number(event.logIndex),
-                valueUSD: usd.reduce((acc, token) => acc + Number(token.valueUSD), 0),
-                payload: {
-                    tokens: usd,
-                },
-            };
-        }),
-    ).catch((e) => {
-        console.error('Error preparing DB entries', e);
-        return [];
-    });
+    // V3 vault join/exit amounts are in wei
+    const allTokens = await prisma.prismaToken.findMany({ where: { chain } });
+    const dbEntries = events.map((event) => ({
+        vaultVersion,
+        id: event.id, // tx + logIndex
+        tx: event.transactionHash,
+        type: event.type === 'Join' ? PoolEventType.JOIN : PoolEventType.EXIT,
+        poolId: event.pool.id,
+        chain: chain,
+        userAddress: event.user.id,
+        blockNumber: Number(event.blockNumber),
+        blockTimestamp: Number(event.blockTimestamp),
+        logIndex: Number(event.logIndex),
+        valueUSD: 0,
+        payload: {
+            tokens: event.pool.tokens.map((token) => ({
+                address: token.address,
+                amount: formatUnits(
+                    BigInt(event.amounts[token.index]),
+                    allTokens.find((t) => t.address === token.address)?.decimals || 18,
+                ),
+                valueUSD: 0,
+            })),
+        },
+    }));
```

Comment:
this could probably live in a `joinExitTransformer` and reused for v2 and v3

Reply:
true, extracting that, although v2 is a bit different